### PR TITLE
Remove isPrimitiveValueTypeClass

### PR DESCRIPTION
**Pull request by a7ehuo**
Also updated code comments where the old primitive value type is referenced

Related: #18157

### CODE REVIEW

**Comment by a7ehuo** [comment]
@hzongaro May I ask you to review this change? Thank you!

File: runtime/compiler/optimizer/EscapeAnalysis.cpp [hunk 1]
```diff
@@ -1208,11 +1209,10 @@ int32_t TR_EscapeAnalysis::performAnalysisOnce()
...
-
-               if (TR::Compiler->cls.isPrimitiveValueTypeClass(clazz))
+               if (clazz && comp()->fej9()->getNullRestrictedArrayClassFromComponentClass(clazz))
                   {
                   if (trace())
-                     traceMsg(comp(), "   Fail [%p] because array has primitive value type elements\n", candidate->_node);
+                     traceMsg(comp(), "   Fail [%p] because array could be null-restricted array\n", candidate->_node);
                   rememoize(candidate);
                   _candidates.remove(candidate);
                   }
```
Comment (hzongaro) [inline runtime/compiler/optimizer/EscapeAnalysis.cpp:1218]: I think this entire block can go away.  The array allocation will always be a nullable array.

File: runtime/compiler/optimizer/J9ValuePropagation.cpp [hunk 2]
```diff
@@ -1096,7 +1096,7 @@ J9::ValuePropagation::constrainRecognizedMethod(TR::Node *node)
             }
 
          if (!canTransformIdentityArrayElementLoadStoreUseTypeHint &&
-             TR::Compiler->cls.isPrimitiveValueTypeClass(hintComponentClass))
+             TR::Compiler->cls.isArrayNullRestricted(comp(), typeHintClass)) //TODO: typeHintClass eventually should be the null-restricted array class
```
Comment (hzongaro) [inline runtime/compiler/optimizer/J9ValuePropagation.cpp:1099]: I would suggest making this `TODO` into a `TODO-VALUETYPE`

File: runtime/compiler/optimizer/EscapeAnalysis.cpp [hunk 3]
```diff
@@ -1191,8 +1191,9 @@ int32_t TR_EscapeAnalysis::performAnalysisOnce()
          {
          // Array Candidates for contiguous allocation that have unresolved
          // base classes must be rejected, since we cannot initialize the array
-         // header.  If the component type is a primitive value type, reject the array
-         // as we can't initialize the elements to the default value yet.
+         // header
+         // If the null-restricted array exists, reject the array as we can't
+         // initialize the elements to the default value yet.
```
[the remark this answers is not in the document]
Comment (hzongaro) [reply runtime/compiler/optimizer/EscapeAnalysis.cpp:1196]: > that we need to handle allocation of null-restricted arrays.

Or I should say, "that will likely want to allow for stack allocation of null-restricted arrays"

File: runtime/compiler/optimizer/EscapeAnalysis.cpp (hunk 1 above)
Comment (a7ehuo) [reply runtime/compiler/optimizer/EscapeAnalysis.cpp]: Addressed in [7bc96d5](https://github.com/eclipse-openj9/openj9/compare/9fd2f09e62d10c0d31fb534bace077e933045ec4..7bc96d56b54a0663b220ede073dc58ef195f2dbe)

File: runtime/compiler/compile/J9Compilation.cpp [hunk 4]
```diff
@@ -727,17 +727,17 @@ J9::Compilation::canAllocateInline(TR::Node* node, TR_OpaqueClassBlock* &classIn
...
          return -1;
 
-      // Arrays of null-restricted (a.k.a, primitive value type) classes must have all their elements initialized
-      // with the default value of the component type.  For now, prevent inline allocation of them.
-      //
-      if (areValueTypesEnabled && TR::Compiler->cls.isPrimitiveValueTypeClass(reinterpret_cast<TR_OpaqueClassBlock*>(clazz)))
+      // Arrays of null-restricted classes must have all their elements initialized
+      // with the default value of the component type.
+      // If null-restricted array class exists, prevent inline allocation of them.
+      if (areValueTypesEnabled && self()->fej9()->getNullRestrictedArrayClassFromComponentClass(TR::Compiler->cls.convertClassPtrToClassOffset(clazz)))
```
[the remark this answers is not in the document]
Comment (a7ehuo) [reply runtime/compiler/compile/J9Compilation.cpp]: Addressed in [7bc96d5](https://github.com/eclipse-openj9/openj9/compare/9fd2f09e62d10c0d31fb534bace077e933045ec4..7bc96d56b54a0663b220ede073dc58ef195f2dbe)

File: runtime/compiler/optimizer/J9ValuePropagation.cpp (hunk 2 above)
Comment (a7ehuo) [reply runtime/compiler/optimizer/J9ValuePropagation.cpp]: Addressed in [7bc96d5](https://github.com/eclipse-openj9/openj9/compare/9fd2f09e62d10c0d31fb534bace077e933045ec4..7bc96d56b54a0663b220ede073dc58ef195f2dbe)

**Comment by a7ehuo** [comment]
@hzongaro All comments are addressed in [7bc96d5](https://github.com/eclipse-openj9/openj9/compare/9fd2f09e62d10c0d31fb534bace077e933045ec4..7bc96d56b54a0663b220ede073dc58ef195f2dbe). Ready for another view. Thanks!

File: runtime/compiler/env/J9ClassEnv.hpp [hunk 5]
```diff
@@ -145,7 +144,7 @@ class OMR_EXTENSIBLE ClassEnv : public OMR::ClassEnvConnector
     * \brief
     *    Checks whether instances of the specified class can be trivially initialized by
     *    "zeroing" their fields.
-    *    In the case of OpenJ9, this tests whether any field is of a primitive value type that
+    *    In the case of OpenJ9, this tests whether any field is of null-restricted field that
```
Comment (hzongaro) [inline runtime/compiler/env/J9ClassEnv.hpp:147]: Typo:  "any field is of null-restricted field" - I think that should read "any field is of a null-restricted type" or something like that.

**Review by hzongaro — APPROVED** [review]
I think the changes look good.  I've noted one typo, but will approve the changes under the assumption that you will correct that.

File: runtime/compiler/env/J9ClassEnv.hpp (hunk 5 above)
Comment (a7ehuo) [reply runtime/compiler/env/J9ClassEnv.hpp]: Addressed in [ab38690](https://github.com/eclipse-openj9/openj9/compare/7bc96d56b54a0663b220ede073dc58ef195f2dbe..ab3869094902f9ea93a5eb7b83504176ea6ee906)

**Comment by hzongaro** [comment]
Jenkins test sanity.functional,sanity.openjdk xlinux,alinux jdk8,jdk21

**Comment by hzongaro** [comment]
Jenkins test sanity.functional,extended.functional xlinuxval,alinuxval,zlinuxval,plinuxval jdknext

**Pull request merged**